Sources/SwiftParser/Declarations.swift

```diff
@@ -1692,11 +1697,28 @@ extension Parser {
       modifier = nil
     }
 
+    if yielding {
+      _ = self.expect(TokenSpec(.yielding))
+    }
+
     let (unexpectedBeforeIntroducer, introducer) = self.expect(kind.spec)
+
+    // Map `yielding borrow` => `read`, etc.
```

Comment:
This breaks round-tripping, so I can't do this mapping here.

Reply:
Implemented `yielding` as a modifier for syntactic purposes.   ASTGen will be more interesting.